velox/vector/FlatVector.cpp

```diff
@@ -498,5 +498,42 @@ void FlatVector<StringView>::copyRanges(
   }
 }
 
+// For strings, we also verify if they point to valid memory locations inside
+// the string buffers.
+template <>
+void FlatVector<StringView>::validate(
+    const VectorValidateOptions& options) const {
+  SimpleVector<StringView>::validate(options);
+  auto byteSize = BaseVector::byteSize<StringView>(BaseVector::size());
+  if (byteSize == 0) {
+    return;
+  }
+  VELOX_CHECK_NOT_NULL(values_);
+  VELOX_CHECK_GE(values_->size(), byteSize);
+  auto rawValues = values_->as<StringView>();
```

Comment:
nit: ensure rawValues is not null (in case of badly formed string vectors)..

Reply:
L511 should ensure values_ is not null and a Buffer cannot have null data_ member if it has non-zero size (confirmed by L512). Should we add checks for validity of the Buffers?

Reply:
What kind of checks do you have in mind ?